File: frontend/src/types/stats.ts
export type CategorySummaryItem = {
  category_id: number;
  category_name: string;
  category_limit: number;
  spent_amount: number;
  remaining_amount: number;
  usage_percent: number;
  is_limit_exceeded: boolean;
};

export type MonthOverviewResponse = {
  month: string;
  budget_id: number | null;
  total_budget: number;
  total_spent: number;
  remaining_budget: number;
  budget_usage_percent: number;
  category_count: number;
  expense_count: number;
  exceeded_categories_count: number;
  categories: CategorySummaryItem[];
};

File: backend/app/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class CategorySummaryItem(BaseModel):
    category_id: int
    category_name: str
    category_limit: float
    spent_amount: float
    remaining_amount: float
    usage_percent: float
    is_limit_exceeded: bool


class MonthOverviewResponse(BaseModel):
    month: date
    budget_id: int | None
    total_budget: float
    total_spent: float
    remaining_budget: float
    budget_usage_percent: float
    category_count: int
    expense_count: int
    exceeded_categories_count: int
    categories: list[CategorySummaryItem]